packages/docs/docs/getting-started/code-block.mdx

```diff
@@ -0,0 +1,249 @@
+
+```tsx
+import {CodeBlock, lines} from '@motion-canvas/2d/lib/components/CodeBlock';
+
+const codeRef = createRef();
+yield view.add(
+  <CodeBlock
+    ref={codeRef}
+    code={`
+      // first line
+      // second line
+      // third line`}
+  />,
+);
+
+// second line only
+yield * codeRef().selection(line(1));
+// second and third line (line 1 to line 2)
+yield * codeRef().selection(line(1, 2));
+```
+
+To remove a selection, you must set the selected range from `[0, 0]` to
+`[Infinity, Infinity]` to select all text. This is most easily performed by
+setting the selection to `lines(0, Infinity)`.
+
+### Animating code
+
+You may insert, remove, or edit the displayed code, any of which will animate
+the code to its new state. All of these changes are performed with the `edit`
+method on a `CodeBlock` instance.
+
+To insert code, start by using `createRef` to store your `CodeBlock` instance
+for future edits. Then call `edit` with an imbedded `insert` call to add the new
+code.
+
+```tsx
+import {CodeBlock, insert} from '@motion-canvas/2d/lib/components/CodeBlock';
+import {createRef} from '@motion-canvas/core/lib/utils';
+
+const codeRef = createRef<CodeBlock>();
+
+yield view.add(<CodeBlock ref={codeRef} code={`var myBool;`} />);
+
+// duration of 1.2 seconds
+yield * codeRef().edit(1.2)`var myBool${insert(' = true')};`;
+```
+
+```tsx
+var myBool;
+// will animate to
+var myBool = true;
+```
+
+Removing code is similar, only with the provided code being removed during the
+animation.
+
+```tsx
+import {CodeBlock, remove} from '@motion-canvas/2d/lib/components/CodeBlock';
+
+yield view.add(<CodeBlock ref={codeRef} code={`var myBool = true;`} />);
+
+yield * codeRef().edit(1.2)`var myBool${remove(' = true')};`;
+```
+
+```tsx
+var myBool = true;
+// will animate to
+var myBool;
+```
+
+Finally, replacing code combines removal and insertion into one call.
+
+```tsx
+yield view.add(<CodeBlock ref={codeRef} code={`var myBool = true;`} />);
+
+yield * codeRef().edit(1.2)`var myBool = ${edit('true', 'false')};`;
+```
+
+```tsx
+var myBool = true;
+// will animate to
+var myBool = false;
+```
+
+Editing code will update your selection to highlight the changes. If you would
+like to retain your selection through an animation, use `edit(duration, false)`.
+
+```tsx
+yield view.add(<CodeBlock ref={codeRef} code={`var myBool;`} />);
+
+// note the second argument to edit
+yield * codeRef().edit(1.2, false)`var myBool${insert(' = true')};`;
+```
+
+You may apply multiple edits to a code block over the course of a video, each
+modifying the prior code to a new state.
+
+```tsx
+yield view.add(<CodeBlock ref={codeRef} code={`var myBool;`} />);
+
+yield * codeRef().edit(1.2)`var myBool${insert(' = true')};`;
+yield * waitFor(1);
+yield * codeRef().edit(1.2)`var myBool = ${edit('true', 'false')};`;
+yield * waitFor(1);
+yield * codeRef().edit(1.2)`var myBool${remove(' = false')};`;
+```
+
+```tsx
+// the displayed code will animate through
```

Comment:
Would it make sense to just pull this out between the code blocks?

Reply:
I'm not sure. The rules for writing are vague when it comes to switching to something like math or code, and I've waffled back and forth on my own rules.

For code, I have a soft rule that sentences which include a code block should be comments in the code, but _starting_ a sentence outside the block and having the code conclude the sentence isn't that bad. I'll try it here.